free-stuff-matrixbot.py

```diff
@@ -1,23 +1,41 @@
 import asyncio
 import json
-import requests
-
+import logging
 from os import environ, path
 
+import requests
 from nio import AsyncClient
 
 
+def default_if_empty(string, default) -> str:
```

Comment:
This is not needed when replacing this custom implementation with [`os.getenv`](https://docs.python.org/3/library/os.html#os.getenv) (see suggestions).

Also in Python you'd write something like `if "" == string` or `if not string` rather than using `__eq__` which is used by `==` (and its negation).

Reply:
The part regarding how to check if the string is empty is fine for me. (I got this from here https://stackoverflow.com/a/55747410, but it's overkill, as stated in a comment there, too. I'll switch to the solution from the accepted answer `if not string`.)

But `os.getenv` allows empty values and I specifically want to use a default in that case.

Reply:
Or wait, no, I don't. It should be possible to build all the configuration (`docker-compose`, Helm Chart`, …) in a way, so that the environment variables are not just empty, but are unset.